File: examples/python/extract_rpm.py
"""Extract an RPM's payload to a directory."""

import argparse

from rpm_rs import Package

parser = argparse.ArgumentParser(description=__doc__)
parser.add_argument("rpm", help="Path to the RPM file")
parser.add_argument("destination", nargs="?", default=".", help="Output directory")
args = parser.parse_args()

dest = args.destination
pkg = Package.open(args.rpm)
nevra = pkg.metadata.nevra()
print(f"Extracting {nevra} to {dest}/")

for f in pkg.files():
    print(f"  {f.metadata.path}")

pkg.extract(dest)

File: examples/extract-rpm.rs
use clap::Parser;
use rpm::Package;
use std::path::PathBuf;

/// Extract an RPM's payload to a directory.
#[derive(Parser)]
#[command(version)]
struct Args {
    /// Path to the RPM file
    rpm: PathBuf,

    /// Destination directory (defaults to current directory)
    #[arg(short, long, default_value = ".")]
    dest: PathBuf,
}

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let args = Args::parse();
    let pkg = Package::open(&args.rpm)?;

    let nevra = pkg.metadata.get_nevra()?;
    println!("Extracting {nevra} to {}/", args.dest.display());

    for file in pkg.files()? {
        let file = file?;
        println!("  {}", file.metadata.path.display());
    }

    pkg.extract(&args.dest)?;

    Ok(())
}
